src/typeChecker/__tests__/source3Typed.test.ts

```diff
@@ -0,0 +1,181 @@
+import { parseError } from '../..'
+import { mockContext } from '../../mocks/context'
+import { parse } from '../../parser/parser'
+import { Chapter, Variant } from '../../types'
+
+let context = mockContext(Chapter.SOURCE_3, Variant.TYPED)
+
+beforeEach(() => {
+  context = mockContext(Chapter.SOURCE_3, Variant.TYPED)
+})
+
+describe('array type', () => {
```

Comment:
What about more than 1-dimensional arrays? The current type system doesn't seem to work.

Reply:
Thanks for the catch! Will add support + tests for 2D+ arrays.